### ISSUE TITLE
setTilePosition issue

### ISSUE THREAD

**Issue by Mteuahasan** [Annoraaq/grid-engine#111]
Hi there,

I'm facing an issue trying to use the setTilePosition method.
Here is the context : I'm making a multiplayer project with socket.io. For this project, I would like Phaser to run even if the browser tab is not focus but it seems to be impossible. So, in order to tackle this, I'm requesting the current position of all players on window's focus event and then I moveTo my other players. BUT, if I wasn't focusing the game's browser tab, when I come back at it, the other players start their movement just now.

I first tried to speed up the other players but I don't managed to succeed. Then I walked into this [issue](https://github.com/Annoraaq/grid-engine/issues/109) where the method `setTilePosition` is mentioned so I tried this :

```
gridEngine.gridCharacters
     .get(onlineUser.id)
      .setTilePosition(
          new Phaser.Math.Vector2(
            onlineUser.coordinates.x,
            onlineUser.coordinates.y
          )
     )
```

It seems to be hacky right ? Anyway, this is working randomly.
Looking at the sources, it seems that the method check whether the player is moving so I tried to call `stopMovement` before `setTilePosition` and I got [this error](https://ibb.co/52fZWCj).

Thank you and sorry the issue is quite verbose. Happy to provide more information if needed 🙂

**Comment by Mteuahasan** [comment]
I managed to have something working by calling `gridEngine.gridCharacters.get(charId).stopMoving()` before `setTilePosition` but it seems quite hacky

**Comment by Annoraaq** [comment]
Hi @Mteuahasan,

Maybe a couple of general information about the used methods first. They might already help you along:

The setTilePosition() method is not yet part of the public API, but I will expose it soon. So you can use it for now, but I would recommend to switch to the official one once I exposed it. However, the idea behind it was that it might mess up things to reposition a moving char. But I think I could also remove that restriction. It would probably not break things. But then if chars are moving they won't be after repositioning.

The stopMovement() and stopMoving() methods are not recommended to use because they are only private methods that are used by the internal movement logic. There names and functionality might change any time and also the code is not designed for them to be called from outside of the plugin. Therefore there is a good chance that you are messing up things in the plugin when you call these.

I think I still don't understand why you need to call the `setTilePosition` method on moving chars, but would it solve your problem if you could call it on moving chars as well as on standing ones?

**Comment by Mteuahasan** [comment]
Hi @Annoraaq,

Thanks for the explaination. I won't use `stopMovement` and `stopMoving` then.
I'm not sure I can explain better than I did without a demo but your proposition would definitely solve the problem, I did this this morning in the unminified version of the plugin and that worked like a charm.

**Comment by Annoraaq** [comment]
Great to hear that. I will implement it like this then.

Btw: There is a bug in `stopMovement()` I am fixing it.
But that method would probably not solve your problem because it will stop any automated movement but only if the character reaches the next tile. Otherwise it would be possible for a character to stand between tiles which is against the nature of the grid physics.

**Comment by Mteuahasan** [comment]
Thank you !
Alright copy that. I won't need to use it anyway if you implement the discussed behavior.

**Comment by Annoraaq** [comment]
I just released version 1.13.0 that should implement the desired behaviour via `setPosition()` method and also fixes `stopMovement()`. Let me know if that solves your problem :)

**Comment by Mteuahasan** [comment]
That totally solved it ! Thank you
